### PR TITLE
refactor: add validate to command/query level

### PR DESCRIPTION
**Pull request by Alan-MQ**
I made some changes on the catlog_write_log_service and If you think it's the right way to do that I will then do the rest
btw there is some changes on the go.mod file(not the new module) please check that too~~
thank you for your time.

### CODE REVIEW

**Comment by mehdihadeli** [comment]
Hi, Thanks for your contribution.
There is CI/Build error in the github action pipeline, maybe the reason is removing `google.golang.org/genproto`. Revert this change in all go.mod file and the check for passing CI/Build

**Comment by Alan-MQ** [comment]
> Hi, Thanks for your contribution. There is a [CI/Build error](https://github.com/mehdihadeli/go-ecommerce-microservices/actions/runs/6021096488/job/16336941482?pr=74#step:5:46) in the GitHub action pipeline, maybe the reason is removing `google.golang.org/genproto` and `mergo` pkg (for now I suggest don't touch the packages, for refactoring pkg and removing useless pkg it's better a separated pull request). Revert this change in all go.mod file and the check for passing CI/Build

ok  I will dod that.

**Comment by mehdihadeli** [comment]
> > Hi, Thanks for your contribution. There is a [CI/Build error](https://github.com/mehdihadeli/go-ecommerce-microservices/actions/runs/6021096488/job/16336941482?pr=74#step:5:46) in the GitHub action pipeline, maybe the reason is removing `google.golang.org/genproto` and `mergo` pkg (for now I suggest don't touch the packages, for refactoring pkg and removing useless pkg it's better a separated pull request). Revert this change in all go.mod file and the check for passing CI/Build
> 
> ok I will do that.

thanks :)

**Comment by Alan-MQ** [comment]
> > > Hi, Thanks for your contribution. There is a [CI/Build error](https://github.com/mehdihadeli/go-ecommerce-microservices/actions/runs/6021096488/job/16336941482?pr=74#step:5:46) in the GitHub action pipeline, maybe the reason is removing `google.golang.org/genproto` and `mergo` pkg (for now I suggest don't touch the packages, for refactoring pkg and removing useless pkg it's better a separated pull request). Revert this change in all go.mod file and the check for passing CI/Build
> > 
> > 
> > ok I will do that.
> 
> thanks :) Also, please update your fork branch, with upstream branch because, there is a small change in the go.mod file in pkg module.

ok, I will revert them and try to make it work, but I can't trigger CI process, should I create a new workflow on my forked project?

**Comment by mehdihadeli** [comment]
I changed permissions for showing github action status for other, for your next change you should see the pipeline status here, for now there is an error in the unit test

**Comment by mehdihadeli** [comment]
> > I changed permissions for showing github action status for others, for your next change you should see the pipeline status here, for now there is an error in the unit test
> 
> ok sure, I will manage to pass that unit test(good one), so every time I commit will trigger the ci right?

Yes, after each push the CI pipeline will trigger, and you can check its result. I recommend running unit tests locally with a [make file](https://github.com/mehdihadeli/go-ecommerce-microservices/blob/main/Makefile#L51) that I prepared locally. For example, in root of project just type `make unit-test` after installing make in your OS, and it will run all unit-tests locally

**Comment by Alan-MQ** [comment]
> 

thank you I will try that make unit-test.

File: internal/services/catalog_read_service/internal/products/features/get_product_by_id/v1/queries/get_product_by_id.go [hunk 1]
```diff
@@ -8,6 +8,8 @@ type GetProductById struct {
 	Id uuid.UUID `validate:"required"`
 }
 
+// TODO
+// this function seems doesn't validate if the id is empty, should I refactor this func and handle the error where NewGetProductByID called?
```
Comment (mehdihadeli) [inline internal/services/catalog_read_service/internal/products/features/get_product_by_id/v1/queries/get_product_by_id.go:12]: Yes, it is better we validate the ID for empty value

File: internal/services/catalog_read_service/internal/products/features/searching_products/v1/endpoints/search_products_endpoint.go [hunk 2]
```diff
@@ -79,7 +79,9 @@ func (ep *searchProductsEndpoint) handler() echo.HandlerFunc {
 			ListQuery:  request.ListQuery,
 		}
 
-		if err := ep.Validator.StructCtx(ctx, query); err != nil {
+		// Am I right?  this one is a validator.StructCtx with   ctx, can I just make it to query.Validate?
+		// if err := ep.Validator.StructCtx(ctx, query); err != nil {
+		if err := query.Validate(); err != nil {
```
Comment (mehdihadeli) [inline internal/services/catalog_read_service/internal/products/features/searching_products/v1/endpoints/search_products_endpoint.go:84]: It is fine, `ep.validator.StructCtx` was for checking validations tags by `go-playground/validator`. Now we could remove it

File: internal/services/catalog_write_service/internal/products/features/getting_products/v1/queries/get_products.go [hunk 3]
```diff
@@ -14,6 +14,9 @@ type GetProducts struct {
 func NewGetProducts(query *utils.ListQuery) (*GetProducts, error) {
 	q := &GetProducts{ListQuery: query}
 
+	// TODO
+	// since there is no validate tag on ListQuery,
+	// maybe we should just remove the next line ?
```
Comment (mehdihadeli) [inline internal/services/catalog_write_service/internal/products/features/getting_products/v1/queries/get_products.go:19]: Yes please do that

**Comment by mehdihadeli** [comment]
Please also update README file, for adding ozz

**Comment by Alan-MQ** [comment]
> Please also update README file, for adding ozz

yeah, I will add that after I fix that linting stuff.

**Comment by Alan-MQ** [comment]
my labeler action keep failing, does it matter? seems like some kind httpResource is not accessbile~~

**Review by mehdihadeli — APPROVED** [review]

**Comment by mehdihadeli** [comment]
> my labeler action keep failing, does it matter? seems like some kind httpResource is not accessbile~~

Yes, `Labler action` has some bug, Also my `linting action` is very strict I should make it simpler, no problem for you.

**Comment by mehdihadeli** [comment]
Thanks for this pull request, and this enhancement, I appreciate about that :)
Just one note after resolving my review, please resolve that review with resolve button, it will look better 👍🏻

**Comment by Alan-MQ** [comment]
> Thanks for this pull request, and this enhancement, I appreciate about that :) Just one note after resolving my review, please resolve that review with resolve button, it will look better 👍🏻

yeah, I didn't know github have that function~~

**Pull request merged**